Automattic/pocket-casts-android
# Referrals: Get referral code from endpoint

## Description
This gets referral code using the endpoint while displaying the send guest pass screen.

## Testing Instructions

1. Switch to "debug" build variant
2. Launch the app and login as a paid user
3. Go to `Profile` tab
4. Tap on gift icon
5. Notice that send guest pass screen is shown
6. Tap "Share Guest Pass" and copy the share text
7. ✅ Notice that there is a real referral code appended to the share URL
8. Close the screen
9. Disable internet
10. Tap on gift icon again
11. ✅ Notice that you see an error screen (this is just a placeholder)
12. Enable internet
13. Tap try again
14. ✅ Notice that data is loaded again.


## Checklist
 
#### I have tested any UI changes...
